From 6fd1f9181aa2cbe18e25054b88617d9882201637 Mon Sep 17 00:00:00 2001
From: Zhaoyang Liu <zhaoyang@codeaurora.org>
Date: Mon, 25 Dec 2017 10:41:19 +0800
Subject: [PATCH] defconfig: Enable 80211AD WiFi driver support

soc: qcom: Support IEEE802.11AD wil6210 device tree.
defconfig: Enable MSM_11AD platform driver.
Support IEEE80211AD WiFi on MSM8996 AUTO platform,
msm_11ad_proxy module is buildin and wil6210 module
works as pcie wifi driver.

Signed-off-by: Zhaoyang Liu <zhaoyang@codeaurora.org>
---
 arch/arm/boot/dts/qcom/msm8996.dtsi        | 4 ++--
 arch/arm64/configs/msm-auto-perf_defconfig | 1 +
 arch/arm64/configs/msm-auto_defconfig      | 1 +
 3 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/qcom/msm8996.dtsi b/arch/arm/boot/dts/qcom/msm8996.dtsi
index 3b067e6..847a425 100644
--- a/arch/arm/boot/dts/qcom/msm8996.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8996.dtsi
@@ -2399,7 +2399,7 @@
 
 	wil6210: qcom,wil6210 {
 		compatible = "qcom,wil6210";
-		qcom,pcie-parent = <&pcie1>;
+		qcom,pcie-parent = <&pcie0>;
 		qcom,wigig-en = <&tlmm 94 0>;
 		qcom,sleep-clk-en = <&pm8994_gpios 18 0>;
 		qcom,msm-bus,name = "wil6210";
@@ -2408,7 +2408,7 @@
 		qcom,msm-bus,vectors-KBps =
 			<100 512 0 0>,
 			<100 512 600000 800000>; /* ~4.6Gbps (MCS12) */
-		status = "disabled";
+		qcom,smmu-support;
 	};
 
 	qcom,ssc@1c00000 {
diff --git a/arch/arm64/configs/msm-auto-perf_defconfig b/arch/arm64/configs/msm-auto-perf_defconfig
index bb0b8f7..4541a00 100644
--- a/arch/arm64/configs/msm-auto-perf_defconfig
+++ b/arch/arm64/configs/msm-auto-perf_defconfig
@@ -286,6 +286,7 @@ CONFIG_USB_USBNET=y
 CONFIG_WCNSS_MEM_PRE_ALLOC=y
 CONFIG_CNSS_CRYPTO=y
 CONFIG_ATH_CARDS=y
+CONFIG_MSM_11AD=y
 CONFIG_WIL6210=m
 CONFIG_CLD_LL_CORE=y
 CONFIG_BUS_AUTO_SUSPEND=y
diff --git a/arch/arm64/configs/msm-auto_defconfig b/arch/arm64/configs/msm-auto_defconfig
index b72807c..8dfc291 100644
--- a/arch/arm64/configs/msm-auto_defconfig
+++ b/arch/arm64/configs/msm-auto_defconfig
@@ -287,6 +287,7 @@ CONFIG_USB_USBNET=y
 CONFIG_WCNSS_MEM_PRE_ALLOC=y
 CONFIG_CNSS_CRYPTO=y
 CONFIG_ATH_CARDS=y
+CONFIG_MSM_11AD=y
 CONFIG_WIL6210=m
 CONFIG_CLD_LL_CORE=y
 CONFIG_BUS_AUTO_SUSPEND=y
-- 
1.9.1

